File: app/src/main/java/edz/example/com/kotlinapplication/service/ResultBean.java
package edz.example.com.kotlinapplication.service;

import android.text.TextUtils;


/**
 * @author： zcs
 * @time：2019/9/24 on 15:49
 * @description：
 */
public class ResultBean<T> extends AbstractResultBean {
    private String code;
    private String message;
    private T data;

    @Override
    public boolean success() {
        return TextUtils.equals(code,"000000");
    }

    @Override
    public String getCode() {
        return code;
    }

    @Override
    public String getMessage() {
        return message;
    }

    public T getData() {
        return data;
    }


}

File: app/src/main/java/edz/example/com/kotlinapplication/service/ResultBean.kt
package edz.example.com.kotlinapplication.service

import android.text.TextUtils


/**
 * @author： zcs
 * @time：2019/9/24 on 15:49
 * @description：
 */
class ResultBean<T> : AbstractResultBean() {
    override val code: String? = null
    override val message: String? = null
    val data: T? = null

    override fun success(): Boolean {
        return TextUtils.equals(code, "000000")
    }


}
